client/extensions/woocommerce/app/order/create.js

```diff
@@ -34,18 +39,37 @@ import { recordTrack } from 'woocommerce/lib/analytics';
 import { sendOrderInvoice } from 'woocommerce/state/sites/orders/send-invoice/actions';
 
 class Order extends Component {
+	possiblyFetchDefaultCurrency( props ) {
+		const { currencyCode, orderEdits, settingsGeneralLoaded, siteId } = props;
+		if ( siteId ) {
+			if ( ! settingsGeneralLoaded ) {
+				this.props.fetchSettingsGeneral( siteId );
+			} else if ( isEmpty( orderEdits.currency ) ) {
+				this.props.editOrder( siteId, { currency: currencyCode } );
+			}
+		}
+	}
+
 	componentDidMount() {
 		const { siteId } = this.props;
 
 		if ( siteId ) {
 			this.props.editOrder( siteId, {} );
 		}
+
+		this.possiblyFetchDefaultCurrency( this.props );
 	}
 
 	componentWillReceiveProps( newProps ) {
 		if ( this.props.siteId !== newProps.siteId ) {
 			this.props.editOrder( newProps.siteId, {} );
 		}
+
+		this.possiblyFetchDefaultCurrency( newProps );
```

Comment:
This shouldn't be necessary if you're also calling it in `componentDidUpdate`, I think this would lead to calling the function twice

Reply:
You're right! I shouldn't be making network requests in componentWillReceiveProps anyways. Fixed in https://github.com/Automattic/wp-calypso/pull/23568/commits/8a7c403776596e9294f02baf3bf29b7b5c04d140